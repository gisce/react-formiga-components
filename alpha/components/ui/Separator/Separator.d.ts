import { SeparatorProps } from "./Separator.types";
export declare const Separator: (props: SeparatorProps) => import("react/jsx-runtime").JSX.Element;
//# sourceMappingURL=Separator.d.ts.map