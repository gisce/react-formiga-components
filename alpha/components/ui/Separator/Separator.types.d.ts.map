{"version":3,"file":"Separator.types.d.ts","sourceRoot":"","sources":["../../../../src/components/ui/Separator/Separator.types.tsx"],"names":[],"mappings":"AAAA,MAAM,MAAM,cAAc,GAAG;IAC3B,KAAK,CAAC,EAAE,MAAM,CAAC;CAChB,CAAC"}