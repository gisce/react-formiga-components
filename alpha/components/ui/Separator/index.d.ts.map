{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/ui/Separator/index.ts"],"names":[],"mappings":"AAAA,cAAc,aAAa,CAAC"}