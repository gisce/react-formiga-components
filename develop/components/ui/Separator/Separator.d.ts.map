{"version":3,"file":"Separator.d.ts","sourceRoot":"","sources":["../../../../src/components/ui/Separator/Separator.tsx"],"names":[],"mappings":"AAEA,OAAO,EAAE,cAAc,EAAE,MAAM,mBAAmB,CAAC;AAEnD,eAAO,MAAM,SAAS,UAAW,cAAc,4CAQ9C,CAAC"}