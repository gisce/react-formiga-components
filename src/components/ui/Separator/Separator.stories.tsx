import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Separator } from ".";
import { BaseFieldArgTypes } from "@/components/form/Field/BaseField.argTypes";

export default {
  title: "Work in progress/UI/Separator",
  component: Separator,
  argTypes: {
    ...BaseFieldArgTypes,
    locale: { table: { disable: true } },
  },
} as ComponentMeta<typeof Separator>;

const Template: ComponentStory<typeof Separator> = (args) => {
  return <Separator {...args} />;
};

export const Basic = Template.bind({});
Basic.args = {
  label: "Separator label",
};

export const WithoutLabel = Template.bind({});
WithoutLabel.args = {};
