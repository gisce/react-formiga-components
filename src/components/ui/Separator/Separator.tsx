import React from "react";
import { Divider } from "antd";
import { SeparatorProps } from "./Separator.types";

export const Separator = (props: SeparatorProps) => {
  const { label } = props;

  return (
    <Divider orientation="left" className="text-sm">
      {label}
    </Divider>
  );
};
