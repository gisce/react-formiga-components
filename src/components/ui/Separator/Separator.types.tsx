export type SeparatorProps = {
  label?: string;
};
