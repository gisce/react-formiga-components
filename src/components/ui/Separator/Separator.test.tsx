import React from "react";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./Separator.stories";
import matchMediaMock from "@/mocks/matchMedia.mock";

describe("Separator", () => {
  beforeAll(() => {
    matchMediaMock();
  });

  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
});
