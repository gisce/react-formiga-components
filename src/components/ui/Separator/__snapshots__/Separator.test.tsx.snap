// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator Renders Basic story 1`] = `
<body>
  <div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left text-sm"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        Separator label
      </span>
    </div>
  </div>
</body>
`;

exports[`Separator Renders WithoutLabel story 1`] = `
<body>
  <div>
    <div
      class="ant-divider ant-divider-horizontal text-sm"
      role="separator"
    />
  </div>
</body>
`;
